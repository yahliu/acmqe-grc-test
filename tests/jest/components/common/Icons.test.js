/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import React from 'react'
import { mount } from 'enzyme'
import {
  GreenCheckCircleIcon,
  RedExclamationCircleIcon,
  YellowExclamationTriangleIcon,
  GrayOutlinedQuestionCircleIcon,
} from '../../../../src-web/components/common/Icons'
import toJson from 'enzyme-to-json'

describe('test component GreenCheckCircleIcon', () => {
  it('renders as expected', () => {
    const component = mount(
      <GreenCheckCircleIcon />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
  it('renders tooltip as expected', () => {
    const component = mount(
      <GreenCheckCircleIcon tooltip='aaaa' />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

})

describe('test component RedExclamationCircleIcon', () => {
  it('renders as expected', () => {
    const component = mount(
      <RedExclamationCircleIcon />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
  it('renders tooltip as expected', () => {
    const component = mount(
      <RedExclamationCircleIcon tooltip='aaaa' />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})


describe('test component YellowExclamationTriangleIcon', () => {
  it('renders as expected', () => {
    const component = mount(
      <YellowExclamationTriangleIcon tooltip='aaaa' />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
  it('renders tooltip as expected', () => {
    const component = mount(
      <YellowExclamationTriangleIcon />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})

describe('test component GrayOutlinedQuestionCircleIcon', () => {
  it('renders as expected', () => {
    const component = mount(
      <GrayOutlinedQuestionCircleIcon tooltip='aaaa' />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
  it('renders tooltip as expected', () => {
    const component = mount(
      <GrayOutlinedQuestionCircleIcon />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
