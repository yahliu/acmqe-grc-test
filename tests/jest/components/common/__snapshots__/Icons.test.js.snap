// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component GrayOutlinedQuestionCircleIcon renders as expected 1`] = `
<GrayOutlinedQuestionCircleIcon
  tooltip="aaaa"
>
  <mockTooltip
    content={
      <div>
        aaaa
      </div>
    }
  >
    <div
      content={
        <div>
          aaaa
        </div>
      }
      data-testid="mockTooltip"
    >
      <OutlinedQuestionCircleIcon
        color="#6a6e73"
        noVerticalAlign={false}
        size="sm"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="#6a6e73"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
          />
        </svg>
      </OutlinedQuestionCircleIcon>
    </div>
  </mockTooltip>
</GrayOutlinedQuestionCircleIcon>
`;

exports[`test component GrayOutlinedQuestionCircleIcon renders tooltip as expected 1`] = `
<GrayOutlinedQuestionCircleIcon>
  <OutlinedQuestionCircleIcon
    color="#6a6e73"
    noVerticalAlign={false}
    size="sm"
  >
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="#6a6e73"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
      />
    </svg>
  </OutlinedQuestionCircleIcon>
</GrayOutlinedQuestionCircleIcon>
`;

exports[`test component GreenCheckCircleIcon renders as expected 1`] = `
<GreenCheckCircleIcon>
  <CheckCircleIcon
    color="#3e8635"
    noVerticalAlign={false}
    size="sm"
  >
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="#3e8635"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
      />
    </svg>
  </CheckCircleIcon>
</GreenCheckCircleIcon>
`;

exports[`test component GreenCheckCircleIcon renders tooltip as expected 1`] = `
<GreenCheckCircleIcon
  tooltip="aaaa"
>
  <mockTooltip
    content={
      <div>
        aaaa
      </div>
    }
  >
    <div
      content={
        <div>
          aaaa
        </div>
      }
      data-testid="mockTooltip"
    >
      <CheckCircleIcon
        color="#3e8635"
        noVerticalAlign={false}
        size="sm"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="#3e8635"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
          />
        </svg>
      </CheckCircleIcon>
    </div>
  </mockTooltip>
</GreenCheckCircleIcon>
`;

exports[`test component RedExclamationCircleIcon renders as expected 1`] = `
<RedExclamationCircleIcon>
  <ExclamationCircleIcon
    color="#c9190b"
    noVerticalAlign={false}
    size="sm"
  >
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="#c9190b"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
      />
    </svg>
  </ExclamationCircleIcon>
</RedExclamationCircleIcon>
`;

exports[`test component RedExclamationCircleIcon renders tooltip as expected 1`] = `
<RedExclamationCircleIcon
  tooltip="aaaa"
>
  <mockTooltip
    content="aaaa"
  >
    <div
      content="aaaa"
      data-testid="mockTooltip"
    >
      <ExclamationCircleIcon
        color="#c9190b"
        noVerticalAlign={false}
        size="sm"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="#c9190b"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          />
        </svg>
      </ExclamationCircleIcon>
    </div>
  </mockTooltip>
</RedExclamationCircleIcon>
`;

exports[`test component YellowExclamationTriangleIcon renders as expected 1`] = `
<YellowExclamationTriangleIcon
  tooltip="aaaa"
>
  <mockTooltip
    content={
      <div>
        aaaa
      </div>
    }
  >
    <div
      content={
        <div>
          aaaa
        </div>
      }
      data-testid="mockTooltip"
    >
      <ExclamationTriangleIcon
        color="#f0ab00"
        noVerticalAlign={false}
        size="sm"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="#f0ab00"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 576 512"
          width="1em"
        >
          <path
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          />
        </svg>
      </ExclamationTriangleIcon>
    </div>
  </mockTooltip>
</YellowExclamationTriangleIcon>
`;

exports[`test component YellowExclamationTriangleIcon renders tooltip as expected 1`] = `
<YellowExclamationTriangleIcon>
  <ExclamationTriangleIcon
    color="#f0ab00"
    noVerticalAlign={false}
    size="sm"
  >
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="#f0ab00"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 576 512"
      width="1em"
    >
      <path
        d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
      />
    </svg>
  </ExclamationTriangleIcon>
</YellowExclamationTriangleIcon>
`;
